Multisig/Logic/Models/Transaction.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  Transaction.swift
+//  Multisig
+//
+//  Created by Moaaz on 6/1/20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import BigInt
+
+struct Transaction: Decodable, Identifiable, Hashable {
+
+    var id: Int {
+        return hashValue
+    }
+
+    let safe: String?
+    let to: String?
+    let value: String?
+    let data: String?
+    let operation: Int?
+    let gasToken: String?
+    let safeTxGas: Int?
+    let baseGas: Int?
+    let gasPrice: String?
+    let refundReceiver: String?
+    let nonce: Int?
+    let executionDate: Date?
+    let submissionDate: Date?
+    let modified: Date?
+    let blockNumber: Int?
+    let transactionHash: String?
+    let safeTxHash: String?
+    let executor: String?
+    let isExecuted: Bool?
+    let isSuccessful: Bool?
+    let tokenAddress: String?
+    let ethGasPrice: String?
+    let gasUsed: Int?
+    let fee: String?
+    let origin: String?
+    let dataDecoded: DecodedData?
+    let confirmationsRequired: Int?
+    let confirmations: [TrnasactionConfirmation]?
+    let signatures: String?
+    let transfers: [Transfer]?
+    let txType: TransactionType?
+}
+
+extension Transaction {
+
+    func status(safeNonce: Int, safeThreshold: Int) -> TransactionStatus {
+        // tx-es without nonce are external transactions that are
+        // already executed successfully.
+        guard let nonce = nonce else { return .success }
+        let confirmationCount = confirmations?.count ?? 0
+        let threshold = confirmationsRequired ?? safeThreshold
+
+        if isExecuted == true && isSuccessful == true {
+            return .success
+        } else if isExecuted == true && isSuccessful != true {
```

Comment:
do we have unit tests for all these cases?

Reply:
no, we don't, nothing is unit tested